pyomo/contrib/doe/measurements.py

```diff
@@ -25,252 +25,270 @@
+    def _use_identity_variance(self):
+        """Generate the variance dictionary. 
+        """
+        self.variance = {}
+        for name in self.name:
+            self.variance[name] = 1     
+
+    
 
-            return string_name
 
+class DesignVariables(SpecialSet):
+    """
+    Define design variables 
+    """
+    def __init__(self):
+        super().__init__()
 
-    def check_subset(self,subset, throw_error=True, valid_subset=True):
+    def specify(self, self_define_res):
+
+        self.name = super().specify(self_define_res)
+
+    def add_elements(self, var_name, extra_index=None, time_index=[0], values=None, 
+                     upper_bound = None, lower_bound = None):
         """
-        Check if the subset is correctly defined with right name, index and time.
 
-        subset:
-            a ''dict'' where measurement name and index are involved in jacobian calculation
-        throw_error:
-            if the given subset is not a subset of the measurement set, throw error message
+        Parameters
+        -----------
+        var_name: a ``list`` of design var names 
+            extra_index: a ``list`` containing extra indexes except for time indexes 
+                if default (None), no extra indexes needed for all var in var_name
+                if it is a nested list, it is a ``list`` of ``list`` of ``list``, 
+                they are different multiple sets of indexes for different var_name
+                for e.g., extra_index[0] are all indexes for var_name[0], extra_index[0][0] are the first index for var_name[0]
+            time_index: a ``list`` containing time indexes
+                default choice is [0], means this is an algebraic variable
+                if it is a nested list, it is a ``list`` of ``lists``, they are different time set for different var in var_name
         """
-        flatten_subset = subset.flatten_measure_name
-        flatten_timeset = subset.flatten_measure_timeset
-        # loop over subset measurement names
-        for i in flatten_subset:
-            # check if subset measurement names are in the overall measurement names
-            if i not in self.flatten_measure_name:
-                valid_subset = False
-                if throw_error:
-                    raise ValueError('This is not a legal subset of the measurement overall set!')
-            else:
-                # check if subset measurement timepoints are in the overall measurement timepoints
-                for t in flatten_timeset[i]:
-                    if t not in self.flatten_measure_timeset[i]:
-                        valid_subset = False
-                        if throw_error:
-                            raise ValueError('The time of {} is not included as measurements before.'.format(t))
-        return valid_subset
\ No newline at end of file
+        self.name =  super().add_elements(var_name=var_name, extra_index=extra_index, time_index=time_index, values=values, 
```

Comment:
Throughout this entire file, please change the order to lower_bound, upper_bound. This is a de facto convention is most packages.